bin/data_frame_analyzer/CCmdLineParser.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#include "CCmdLineParser.h"
+
+#include <ver/CBuildInfo.h>
+
+#include <boost/program_options.hpp>
+
+#include <iostream>
+
+namespace ml {
+namespace data_frame_analyzer {
+
+const std::string CCmdLineParser::DESCRIPTION = "Usage: data_frame_analyzer [options]\n"
+                                                "Options:";
+
+bool CCmdLineParser::parse(int argc,
+                           const char* const* argv,
+                           std::string& configFile,
+                           std::string& logProperties,
+                           std::string& logPipe,
+                           std::string& inputFileName,
+                           bool& isInputFileNamedPipe,
+                           std::string& outputFileName,
+                           bool& isOutputFileNamedPipe) {
+    try {
+        boost::program_options::options_description desc(DESCRIPTION);
+        desc.add_options()("help", "Display this information and exit")(
```

Comment:
You can make this list of options look a lot nicer by putting `// clang-format off` at the beginning and `// clang-format on` at the end.  All the other `CCmdLineParser.cc` files do this.